### PR TITLE
fix email not included in textsearch

### LINKED ISSUE #1979: Search by email no longer functions
Searching by an email address alone used to show all invoices with that email address associated with it. However as of last version, this search functionality no longer exists *ON NEW ORDERS* only. 

Instead any email search simply returns "There are no invoices matching your criteria."

Example:
Searching for my email shows all results from a few months ago

Searching a new customers email address who only placed orders today shows "There are no invoices matching your criteria."

### PR DESCRIPTION
**Pull request by Kukks**
fixes #1979

### CODE REVIEW

File: BTCPayServer/Services/Invoices/InvoiceRepository.cs [hunk 1]
```diff
@@ -204,7 +205,7 @@ public async Task<InvoiceEntity> CreateInvoiceAsync(string storeId, InvoiceEntit
             textSearch.Add(invoice.Metadata.OrderId);
             textSearch.Add(ToString(invoice.Metadata, null));
             textSearch.Add(invoice.StoreId);
-
+            textSearch.Add(invoice.Metadata.BuyerEmail);
```
Comment (NicolasDorier) [inline BTCPayServer/Services/Invoices/InvoiceRepository.cs:208]: it should be in metadata, why adding it here?

Comment (Kukks) [reply BTCPayServer/Services/Invoices/InvoiceRepository.cs:208]: i checke and the search just doesnt work with only the stringified metadata added to text search

**Comment by britttttk** [comment]
tACK - It works great for email provided on create invoice page and customer email on invoice

**Review by pavlenex — APPROVED** [review]
tACK. Works great & as expected.

**Pull request merged**